src/WebJobs.Script.WebHost/Security/KeyManagement/SimpleKubernetesClient.cs

```diff
@@ -0,0 +1,307 @@
+                    var payload = new
+                    {
+                        kind = isSecret ? "Secret" : "ConfigMap",
+                        apiVersion = "v1",
+                        metadata = new
+                        {
+                            name = name
+                        }
+                    };
+                    using (var createRequest = await GetRequest(HttpMethod.Post, url, payload))
+                    {
+                        var createResponse = await _httpClient.SendAsync(createRequest);
+                        createResponse.EnsureSuccessStatusCode();
+                    }
+                }
+            }
+        }
+
+        private async Task<(string, bool)> GetObjectUrl(string objectName, bool watchUrl = false)
+        {
+            var isSecret = true;
+            if (!objectName.StartsWith("secrets/") && !objectName.StartsWith("configmaps/"))
+            {
+                // assume a secret if no type is specified.
+                objectName = $"secrets/{objectName}";
+            }
+            else if (objectName.StartsWith("configmaps/"))
+            {
+                // This allows the users to manage their keys in configMaps instead of secret.
+                // There is no difference currently in kubernetes between the 2 other than secrets
+                // requiring base64 encoding. If it's a configMap, then we don't need to decode it.
+                isSecret = false;
+            }
+
+            var @namespace = await FileUtility.ReadAsync(NamespaceFile);
+            var url = $"{_environment.GetKubernetesApiServerUrl()}/api/v1/";
+            if (watchUrl)
+            {
+                url += "watch/";
+            }
+            url += $"namespaces/{@namespace}/{objectName}";
+            return (url, isSecret);
+        }
+
+        private async Task<HttpRequestMessage> GetRequest(HttpMethod method, string url, object payload = null, string contentType = null, bool closeConnection = true)
+        {
+            var token = await FileUtility.ReadAsync(TokenFile);
+            const string jsonContentType = "application/json";
+            contentType = contentType ?? jsonContentType;
+
+            var request = new HttpRequestMessage(method, url);
+            request.Headers.Add("Authorization", $"Bearer {token}");
+            request.Headers.Add("Accept", jsonContentType);
+
+            if (closeConnection)
+            {
+                request.Headers.Add("Connection", "close");
+            }
+
+            if (payload != null)
+            {
+                var jsonPayload = JsonConvert.SerializeObject(payload);
+                request.Content = new StringContent(jsonPayload, Encoding.UTF8, contentType);
+            }
+
+            return request;
+        }
+
+        private static async Task<IDictionary<string, string>> GetFromFiles(string path)
+        {
+            var files = await FileUtility.GetFilesAsync(path, "*");
```

Comment:
nit: `var` -> `string` (for files)

Reply:
what is the rule you use for `var` vs typeName?

Reply:
The key thing (for me) is whether or not the type is evident.

If the variable is being assigned from a method call, (e.g. `var foo = SomeMethod()`), I opt to use the type instead of `var` as that significantly improves readability.